src/h3/mod.rs

```diff
@@ -474,6 +482,9 @@ pub enum Event {
 
     /// Stream was closed,
     Finished,
+
+    /// GOAWAY was received.
+    GoAway(u64),
```

Comment:
Instead of adding the "id" to the event, would it make sense to jusr use the `stream_id` we already return from `poll()` for this? It might be confusing to have both.

Reply:
good point, it would also make the C API easier

`poll()` currently says 

    /// On success it returns an [`Event`] as well as the event's source stream
    /// ID. The stream ID can be used when calling [`send_response()`] and
    /// [`send_body()`] when responding to incoming requests. On error the
    /// connection will be closed by calling [`close()`] with the appropriate
    /// error code.

With GOAWAY, it won't be `source stream ID` but I think we can accomodate that without breaking the current API. Let me make a commit that augments that description for review.